### PR TITLE
feat: implements custom GitHub action

### LINKED ISSUE #11: custom github action
This repo should be configured with an `action.yml` file containing metadata for a publicly available custom GitHub Action.

### PR DESCRIPTION
**Pull request by talentedmrjones**
## Added

`action.yml` as metadata for a custom github action
 
## Changed

updated readme with instructions for using the github action, and  better instructions for testing locally.

closes #11

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,44 +2,65 @@
...
-  authorization: "<authorization header>"
 tests:
-- url:
-  method:
+- url: string
+  method: string #a supported HTTP method such as GET, POST, PUT, DELETE, etc...
   expect:
+    status: int #a supported HTTP status code such as 200,201,301,400,404, etc...
+    body: string # optional
+    headers: #optional
```
Comment (sbolel) [inline README.md:25]: Nitpick 
```suggestion
    headers: # optional
```

File: README.md [hunk 2]
```diff
@@ -2,44 +2,65 @@
...
 ```yaml
-commonHeaders: &commonHeaders
-  authorization: "<authorization header>"
 tests:
-- url:
-  method:
+- url: string
+  method: string #a supported HTTP method such as GET, POST, PUT, DELETE, etc...
   expect:
+    status: int #a supported HTTP status code such as 200,201,301,400,404, etc...
```
Comment (sbolel) [inline README.md:23]: Nitpitck 
```suggestion
    status: int # a supported HTTP status code such as 200,201,301,400,404, etc...
```

**Review by devon-berta-aquia — APPROVED** [review]
LGTM

File: README.md [hunk 3]
```diff
@@ -2,44 +2,65 @@
...
+The following example is the minimum configuration required to execute Emberfall tests. This performs a single test against an API endpoint, expecting to receive a 401 unauthorized response:
+```yaml
+tests:
 - url: http://localhost:3000/api/v1/users/current
   method: GET
   expect:
     status: 401
     body: "unauthorized"
-
+```
```
Comment (sbolel) [inline README.md:38]: Is there something going on with the rendering of the yaml here?

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+          tar -C /tmp -xzf emberfall.tgz
+          rm emberfall.tgz
+          sudo mv /tmp/emberfall /usr/bin
+      
+      - name: Start emberfall
+        shell: bash
+        run: |
+          echo "${{ inputs.config }}" > ./emberfallconfig.yml
+          emberfall --config ./emberfallconfig.yml
+
```
Comment (sbolel) [inline action.yml:49]: Nitpick
```suggestion


```

**Review by sbolel — APPROVED** [review]
LGTM with minor nitpick suggestions

File: README.md (hunk 3 above)
Comment (jsosulska) [reply README.md:38]: It may be that a newline is needed to render it correctly on GH, as on L27-29

File: action.yml (hunk 4 above)
Comment (devon-berta-aquia) [reply action.yml:49]: Hmm there should be a new line but should not be extra space characters on the line. Hard to tell without highlighting in the UI.

**Pull request merged**